feature/home/src/commonMain/kotlin/com/tunjid/heron/home/HomeViewModel.kt

```diff
@@ -121,10 +125,16 @@ private fun loadProfileMutations(
 private fun timelineMutations(
     scope: CoroutineScope,
     timelineRepository: TimelineRepository,
+    savedStateDataSource: SavedStateDataSource,
 ): Flow<Mutation<State>> =
     timelineRepository.homeTimelines().mapToMutation { homeTimelines ->
+
+        val lastViewed = savedStateDataSource.savedState.value
```

Comment:
Let's improve on what the AI suggested by using the `combine` operator:

```kotlin
combine(
    // We only care about the very first value emitted
    savedStateDataSource.savedState.take(1),
    timelineRepository.homeTimelines(),
    ::Pair,
) { savedState, homeTimelines -> 
    val tabUri = currentTabUri 
    ?: lastViewed?.takeIf { uri -> 
        homeTimelines.indexOfFirst { it.isPinned && it.uri == uri } >= 0 
    }
    ?: homeTimelines.firstOrNull()?.uri
    
    copy(
        currentTabUri = tabUri,
        ...
    )
}
```

This lets us:
* Preserve the last in memory value.
* Use the last saved value if it still exists in the pinned home timelines (the user may have update timelines in the browser)

Reply:
Resolved.